File: app/src/main/java/com/lutech/flashlight/Notification/MyApplication.java
package com.lutech.flashlight.Notification;

import android.app.Application;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.os.Build;
import android.util.Log;

public class MyApplication extends Application {
    public static final String CHANNEL_ID = "channel_service_example";
//    public  static final String CHANNEL_ID2 = "channel_service_example2";

    @Override
    public void onCreate() {
        super.onCreate();
        createChannelNotification();
    }

    private void createChannelNotification() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            NotificationChannel channel = new NotificationChannel(CHANNEL_ID, "Channel Service Example", NotificationManager.IMPORTANCE_DEFAULT);
            NotificationManager manager = (NotificationManager) getSystemService(NOTIFICATION_SERVICE);
            manager.createNotificationChannel(channel);
            if (manager != null) {
                manager.createNotificationChannel(channel);
                Log.d("====>32456", "create noti: ");
            }
        }
    }

//    private void createChannelNotification2() {
//        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
//            NotificationChannel channel = new NotificationChannel(CHANNEL_ID2,"Channel Service Example", NotificationManager.IMPORTANCE_DEFAULT);
//            NotificationManager manager = getSystemService(NotificationManager.class);
//            manager.createNotificationChannel(channel);
//            if (manager != null){
//                manager.createNotificationChannel(channel);
//                Log.d("====>32456", "create noti: ");
//            }
//        }
//    }


}

File: app/src/main/java/com/lutech/flashlight/ads/MyApplication.kt
package com.lutech.flashlight.ads

import android.app.Activity
import android.app.Application
import android.os.Bundle
import android.util.Log
import com.adjust.sdk.Adjust
import com.adjust.sdk.AdjustConfig
import com.adjust.sdk.BuildConfig
import com.adjust.sdk.LogLevel
import com.google.android.gms.ads.MobileAds

class MyApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        try {
            Log.d("88888888888", "AppOpenManager  ")
            MobileAds.initialize(
                this
            ) {
                appOpenManager = AppOpenManager(this)
            }
//            OneSignal.setLogLevel(OneSignal.LOG_LEVEL.VERBOSE, OneSignal.LOG_LEVEL.NONE)
//            OneSignal.initWithContext(this)
//            OneSignal.setAppId("c53f48f1-1283-4d2c-82f2-ca7a721868a1")
//            startMyService()
            val appToken = ""
            val environment: String = if (BuildConfig.DEBUG) {
                AdjustConfig.ENVIRONMENT_SANDBOX
            } else {
                AdjustConfig.ENVIRONMENT_PRODUCTION
            }
            val config = AdjustConfig(this, appToken, environment)
            config.setLogLevel(LogLevel.VERBOSE);
            config.setPreinstallTrackingEnabled(true)

            Adjust.onCreate(config)
            registerActivityLifecycleCallbacks(AdjustLifecycleCallbacks())
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }

    private class AdjustLifecycleCallbacks : ActivityLifecycleCallbacks {
        override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) {
        }

        override fun onActivityStarted(activity: Activity) {
        }

        override fun onActivityResumed(activity: Activity) {
            Adjust.onResume()
        }

        override fun onActivityPaused(activity: Activity) {
            Adjust.onPause()
        } //...

        override fun onActivityStopped(activity: Activity) {

        }

        override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) {
        }

        override fun onActivityDestroyed(activity: Activity) {
        }
    }

//    private fun startMyService() {
//        try {
//            if (!Utils.isMyServiceRunning(NotificationService::class.java, applicationContext!!)) {
//                when {
//                    Build.VERSION.SDK_INT >= Build.VERSION_CODES.S -> {
//                        applicationContext.startService(Intent(applicationContext, NotificationService::class.java))
//                    }
//                    Build.VERSION.SDK_INT >= Build.VERSION_CODES.O -> {
//                        applicationContext.startForegroundService(
//                            Intent(
//                                applicationContext,
//                                NotificationService::class.java
//                            )
//                        )
//                    }
//                    else -> {
//                        applicationContext.startService(Intent(applicationContext, NotificationService::class.java))
//
//                    }
//                }
//            }
//        } catch (e: Exception) {
//            e.printStackTrace()
//        }
//    }


    companion object {

        var appOpenManager: AppOpenManager? = null

        var myApplication: MyApplication? = null

        fun getInstance(): MyApplication {
            if (myApplication == null) {
                myApplication = MyApplication()
            }
            return myApplication!!
        }
    }

}